### ISSUE TITLE
uuid.v4() does not produce unique UUIDs by default

### ISSUE THREAD

**Issue by hoylen** [Daegalus/dart-uuid#46]
The `v4()` method can generate duplicate UUIDs, when used with the default configuration (using the `mathRNG` random number generator). The expected behaviour is for UUIDs to be unique.

See attached test.
[uuid_v4_uniqueness_test.txt](https://github.com/Daegalus/dart-uuid/files/4768690/uuid_v4_uniqueness_test.txt)

"Simple, fast," but incorrect is not good!

I can see there could be valid reasons for not using `UuidUtil.cryptoRNG`. But I would argue that **_cryptoRNG_ should be made the default**, and anyone who wants to risk having duplicate UUIDs can knowingly and intentionally do so by changing to _mathRNG_. Most programmers will start with the defaults, and they could unwittingly run into problems without realising it until it is too late.

**Cryptographically secure means something else**

The current documentation on the random number generators don't make this point clear. It only talks about them being "cryptographically strong" vs not cryptographically strong.

In the context of generating UUIDs, the issue of "cryptographically strong" usually refers to whether the UUIDs are suitable for security purposes. For example, the "IP address + time" version UUIDs might be perfectly unique, but their values can be predicted by an attacker. Therefore, a programmer reading that description might choose _mathRNG_, because they only need unique UUIDs but does not need them to be secure.

Actually, does anyone actually want a UUID generator that can produce duplicates? Especially when the probability of duplicates appears to be greater than 1 in 10,000 ?

**Comment by Daegalus** [comment]
Thank you for this bug report. Especially with a reproducible script. 

Theoretically, we shouldn't be getting duplicates with MathRNG either. So I believe there is a bug.

You aren't the first to report it but no one has been able to provide any reproducible example. I haven't run your script yet.

I have also run similar scripts to test up to 10 million, and never had duplicates with MathRNG. So I am curious what is different.

**Comment by hoylen** [comment]
The test doesn't do anything special: it just calls `v4` just 300,000 times in a loop -- which usually produces over 100 duplicates with _mathRNG_. It is a simple test, so I'm surprised that it hasn't been reproduced before... unless there is another factor involved.

I've run the test on macOS 10.15.5 and CentOS 7.7, both with Dart 2.8.4.

**Comment by Daegalus** [comment]
Maybe i had a bug in my own script that tested it, but I managed reproduce it with your script. I made a small adjustment to the `mathRNG` function (i shuffle the resulting list before returning it). 
With this change, it's still fast, but i had no duplicates even up to a 3 million loop. I will be releasing this shortly with a test that will make sure we can do a 3 million loop with no duplicates.

Thank you
